lib/perl/OVH/Bastion/allowdeny.inc

```diff
@@ -490,12 +490,23 @@ sub print_acls {
     my $acls       = $params{'acls'} || [];
     my $reverse    = $params{'reverse'};
     my $hideGroups = $params{'hideGroups'};
-
-    my $printIpLen = $reverse ? 30 : 15;
-    osh_info(
-        sprintf("%-" . $printIpLen . "s %5s %20s %30s %10s %13s %45s %40s %s", "IP", "PORT", "USER", "ACCESS-BY", "ADDED-BY", "ADDED-AT", "EXPIRY?", "COMMENT", "FORCED-KEY?"));
-
-    my @flatArray;
+    my $includes   = $params{'includes'} || [];
+    my $excludes   = $params{'excludes'} || [];
+
+    my $includere = OVH::Bastion::build_re_from_wildcards(wildcards => $includes, implicit_contains => 1)->value;
+    my $excludere = OVH::Bastion::build_re_from_wildcards(wildcards => $excludes, implicit_contains => 1)->value;
+
+    # first, get all the rows we'll print, and fill both the array that will be printed (printRows),
+    # and the one that will be returned as JSON (jsonRows). We also apply the filters here to include/exclude
+    # the requested patterns, if any
+    # also take this opportunity to remember the longest field for each column
+    my @printRows;
+    my @jsonRows;
+    my @columnNames = qw( IP PORT USER ACCESS-BY ADDED-BY ADDED-AT EXPIRY? COMMENT FORCED-KEY );
+    my @printRowsLength;
```

Comment:
As a not, `foreach` + `push` == `map`

`my @printRowsLength = map length, @columnNames;

Reply:
So perly! thx